opm/simulators/flow/BlackoilModel.hpp

```diff
@@ -910,12 +912,29 @@ namespace Opm {
             auto cnvErrorPvFraction = computeCnvErrorPv(B_avg, dt);
             cnvErrorPvFraction /= (pvSum - numAquiferPvSum);
 
-            const double tol_mb  = param_.tolerance_mb_;
             // Default value of relaxed_max_pv_fraction_ is 0.03 and min_strict_cnv_iter_ is 0.
             // For each iteration, we need to determine whether to use the relaxed CNV tolerance.
             // To disable the usage of relaxed CNV tolerance, you can set the relaxed_max_pv_fraction_ to be 0.
-            const bool use_relaxed = cnvErrorPvFraction < param_.relaxed_max_pv_fraction_ && iteration >= param_.min_strict_cnv_iter_;
-            const double tol_cnv = use_relaxed ? param_.tolerance_cnv_relaxed_ :  param_.tolerance_cnv_;
+            bool use_relaxed_cnv = cnvErrorPvFraction < param_.relaxed_max_pv_fraction_ && iteration >= param_.min_strict_cnv_iter_;
+            bool use_relaxed_mb = param_.min_strict_mb_iter_ < 0? false : iteration >= param_.min_strict_mb_iter_;
```

Comment:
Nitpick: space around `?`. However, I think the logic here is a little too complicated, and also slightly flawed. With this code, even if I set the number of strict iterations to 1000, I will still get a relaxed final iteration, but that should only happen with -1. Also, it is better to keep these bools const. I suggest:
```
const bool relax_final_iteration = (param_.min_strict_mb_iter_ < 0) && (iteration + 1 == maxIter);
const bool use_relaxed_mb = relax_final_iteration || (iteration >= param_.min_strict_mb_iter_);
```
And then replace the `if ()` below with
```
if (relax_final_iteration) {
    // Write the message about trying with relaxed tolerances in the final iteration.
}
```

Reply:
I have updated accordingly.